frontend/src/BasicInfo.js

```diff
@@ -3,46 +3,21 @@ import { Card } from '@material-ui/core';
 import Credits from './Credits';
 import TermDates from './TermDates';
 
-class BasicInfo extends React.Component {
-  constructor(props) {
-    super(props);
-
-    this.updateMinCredits = this.updateMinCredits.bind(this);
-    this.updateMaxCredits = this.updateMaxCredits.bind(this);
-    this.updateTermStartDate = this.updateTermStartDate.bind(this);
-    this.updateTermEndDate = this.updateTermEndDate.bind(this);
-  }
-
-  updateMinCredits(minCredits) {
-    this.props.updateMinCredits(minCredits);
-  }
-
-  updateMaxCredits(maxCredits) {
-    this.props.updateMaxCredits(maxCredits);
-  }
-
-  updateTermStartDate(startDate) {
-    this.props.updateTermStartDate(startDate);
-  }
-
-  updateTermEndDate(endDate) {
-    this.props.updateTermEndDate(endDate);
-  }
-  
+class BasicInfo extends React.Component {  
```

Comment:
This component is functionally pure, so you can replace it with a lambda.

https://www.tutorialspoint.com/pure-component-in-react-js

Reply:
Got it, thanks for showing me this!